app/components/base_component.rb

```diff
@@ -1,14 +1,9 @@
 class BaseComponent < ViewComponent::Base
-  def render_in(view_context, &block)
-    render_scripts_in(view_context)
-    super(view_context, &block)
-  end
-
-  def render_scripts_in(view_context)
+  def before_render
     return if @rendered_scripts
     @rendered_scripts = true
-    if view_context.respond_to?(:render_component_script) && self.class.scripts.present?
-      view_context.render_component_script(*self.class.scripts)
+    if helpers.respond_to?(:render_component_script) && self.class.scripts.present?
+      helpers.render_component_script(*self.class.scripts)
```

Comment:
Aside: "render" is a misnomer here, since it won't actually do anything with a return value. It's more of a signal of _intent_ that a script should be rendered. This is also relevant to the fact that this happens before the render of the component's markup, where depending if the implementation were to render those scripts synchronously, the markup isn't yet in the page for the component to be initialized. This works fine for us though, since `render_component_script` is an [alias to `javascript_packs_tag_once`](https://github.com/18F/identity-idp/blob/6b378d8ed28543ac2fbd637d3916bbec7d994551/app/helpers/script_helper.rb#L23).

For future consideration:

- Support return values from `render_component_script`, where the return value is appended to the component's own markup.
- Or, change the name of the method to something which better indicates its behavior as a signal.

Reply:
I'd vote to change the method name, maybe like `script_tag_names` or something?

Reply:
Having implemented it based on WordPress's script handling, maybe adapting its [`wp_enqueue_script`](https://developer.wordpress.org/reference/functions/wp_enqueue_script/) to similar language? e.g. `enqueue_component_scripts` ? A bit jargon-y, though descriptive.

Just wondering if `script_tag_names` might mislead something to think they could _return_ a set of script tags, or otherwise behave similar to [`javascript_tag`](https://api.rubyonrails.org/classes/ActionView/Helpers/JavaScriptHelper.html).

Reply:
I like `enqueue_component_scripts`, maybe related, `register_component_scripts`

Reply:
👍  Updated to `enqueue_component_scripts` in 1727fb8.